README.md
# Terraform EKS Tags

## Terraform Version

```
terraform version
---
Terraform v1.0.6
+ provider registry.terraform.io/hashicorp/aws v3.61.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

## Folder Structure

### **Core**

All the components inside `Core` folder are related with Backbone Network Service eg:

- VPC
- Subnet Configuration
- Internet Gateway
- NAT Gateway
- and others

### **Resources**

All the components inside `Resources` folder are related to AWS Services that related with application deployment, eg:

- EKS
- EC2
- S3
- and others

### **Modules (Terraform Submodules)**

- Official Repositories

  ```
  ./get-official.sh
  -- or --
  make sub-official
  ```

- Community Repositories

  ```
  ./get-community.sh
  -- or --
  make sub-community
  ```

## How-to-Use

- Clone this repository

  ```
  git clone git@gitlab.com:zeroc0d3lab-devops/terraform-eks-tags.git
  ```

- Pull Submodule repository

  ```
  git submodule update --init --recursive
  -- or --
  make sub-all
  ```

- Assume Role for Terraform command

  ```
  aws sts assume-role --role-arn arn-role-to-assume --role-session-name AWSCLI
  ```

- Terraform Execution
  ```
  terraform init
  terraform plan
  terraform apply
  ```


Changelog.md
# Changelog History

## Version 0.1

- Initial commit for terraform-eks-tags

## Version 0.1.1

- Completed core & resources poc from DevOpsCorner.id

## Version 0.1.2

- Added container ubuntu as bastion access


get-community.sh
#!/usr/bin/env sh
# -----------------------------------------------------------------------------
#  Get Community Terraform Submodules
# -----------------------------------------------------------------------------
#  Author     : Dwi Fahni Denni (@zeroc0d3)
#  License    : Apache v2
# -----------------------------------------------------------------------------
set -e

SUBMODULE_TERRAFORM="git@github.com:cloudposse/terraform-aws-route53-alias.git \
    git@github.com:cloudposse/terraform-aws-s3-log-storage.git \
    git@github.com:cloudposse/terraform-null-label.git \
"

PATH_APP=`pwd`
PATH_MODULES=$PATH_APP/modules
COL_RED="\033[22;31m"
COL_GREEN="\033[22;32m"
COL_BLUE="\033[22;34m"
COL_END="\033[0m"

get_time() {
  DATE=`date '+%Y-%m-%d %H:%M:%S'`
}

print_line0() {
  echo "$COL_GREEN=====================================================================================$COL_END"
}

print_line1() {
  echo "$COL_GREEN-------------------------------------------------------------------------------------$COL_END"
}

print_line2() {
  echo "-------------------------------------------------------------------------------------"
}

logo() {
  clear
  print_line0
  echo "$COL_RED'########:'########:'########:::'#######:::'######::::'#####:::'########:::'#######:: $COL_END"
  echo "$COL_RED..... ##:: ##.....:: ##.... ##:'##.... ##:'##... ##::'##.. ##:: ##.... ##:'##.... ##: $COL_END"
  echo "$COL_RED:::: ##::: ##::::::: ##:::: ##: ##:::: ##: ##:::..::'##:::: ##: ##:::: ##:..::::: ##: $COL_END"
  echo "$COL_RED::: ##:::: ######::: ########:: ##:::: ##: ##::::::: ##:::: ##: ##:::: ##::'#######:: $COL_END"
  echo "$COL_RED:: ##::::: ##...:::: ##.. ##::: ##:::: ##: ##::::::: ##:::: ##: ##:::: ##::...... ##: $COL_END"
  echo "$COL_RED: ##:::::: ##::::::: ##::. ##:: ##:::: ##: ##::: ##:. ##:: ##:: ##:::: ##:'##:::: ##: $COL_END"
  echo "$COL_RED ########: ########: ##:::. ##:. #######::. ######:::. #####::: ########::. #######:: $COL_END"
  echo "$COL_RED........::........::..:::::..:::.......::::......:::::.....::::........::::.......::: $COL_END"
  print_line1
}

header() {
  logo
  print_line0
  get_time
  echo "$COL_RED# BEGIN PROCESS..... (Please Wait)  $COL_END"
  echo "$COL_RED# Start at: $DATE  $COL_END\n"
}

footer() {
  print_line0
  get_time
  echo "$COL_RED# Finish at: $DATE  $COL_END"
  echo "$COL_RED# END PROCESS.....  $COL_END\n"
}

submodule_terrafom() {
    print_line2
    get_time
    echo "$COL_BLUE[ $DATE ] ##### Download Official Submodule(s): $COL_END"

    cd $PATH_MODULES
    for SBTR in $SUBMODULE_TERRAFORM
    do
      get_time
      print_line2
      echo "$COL_GREEN[ $DATE ]       $SBTR \n"
      print_line2
      git submodule add $SBTR
      echo ""
    done
}

main() {
  header
  submodule_terrafom
  footer
}

### START HERE ###
main $@


get-official.sh
#!/usr/bin/env sh
# -----------------------------------------------------------------------------
#  Get Official Terraform Submodules
# -----------------------------------------------------------------------------
#  Author     : Dwi Fahni Denni (@zeroc0d3)
#  License    : Apache v2
# -----------------------------------------------------------------------------
set -e

SUBMODULE_TERRAFORM="terraform-aws-alb \
    terraform-aws-autoscaling \
    terraform-aws-eks \
    terraform-aws-elb \
    terraform-aws-iam \
    terraform-aws-rds \
    terraform-aws-route53 \
    terraform-aws-s3-bucket \
    terraform-aws-security-group \
    terraform-aws-vpc \
    terraform-aws-vpn-gateway \
"

PATH_APP=`pwd`
PATH_MODULES=$PATH_APP/modules
COL_RED="\033[22;31m"
COL_GREEN="\033[22;32m"
COL_BLUE="\033[22;34m"
COL_END="\033[0m"

get_time() {
  DATE=`date '+%Y-%m-%d %H:%M:%S'`
}

print_line0() {
  echo "$COL_GREEN=====================================================================================$COL_END"
}

print_line1() {
  echo "$COL_GREEN-------------------------------------------------------------------------------------$COL_END"
}

print_line2() {
  echo "-------------------------------------------------------------------------------------"
}

logo() {
  clear
  print_line0
  echo "$COL_RED'########:'########:'########:::'#######:::'######::::'#####:::'########:::'#######:: $COL_END"
  echo "$COL_RED..... ##:: ##.....:: ##.... ##:'##.... ##:'##... ##::'##.. ##:: ##.... ##:'##.... ##: $COL_END"
  echo "$COL_RED:::: ##::: ##::::::: ##:::: ##: ##:::: ##: ##:::..::'##:::: ##: ##:::: ##:..::::: ##: $COL_END"
  echo "$COL_RED::: ##:::: ######::: ########:: ##:::: ##: ##::::::: ##:::: ##: ##:::: ##::'#######:: $COL_END"
  echo "$COL_RED:: ##::::: ##...:::: ##.. ##::: ##:::: ##: ##::::::: ##:::: ##: ##:::: ##::...... ##: $COL_END"
  echo "$COL_RED: ##:::::: ##::::::: ##::. ##:: ##:::: ##: ##::: ##:. ##:: ##:: ##:::: ##:'##:::: ##: $COL_END"
  echo "$COL_RED ########: ########: ##:::. ##:. #######::. ######:::. #####::: ########::. #######:: $COL_END"
  echo "$COL_RED........::........::..:::::..:::.......::::......:::::.....::::........::::.......::: $COL_END"
  print_line1
}

header() {
  logo
  print_line0
  get_time
  echo "$COL_RED# BEGIN PROCESS..... (Please Wait)  $COL_END"
  echo "$COL_RED# Start at: $DATE  $COL_END\n"
}

footer() {
  print_line0
  get_time
  echo "$COL_RED# Finish at: $DATE  $COL_END"
  echo "$COL_RED# END PROCESS.....  $COL_END\n"
}

submodule_terrafom() {
    print_line2
    get_time
    echo "$COL_BLUE[ $DATE ] ##### Download Official Submodule(s): $COL_END"

    cd $PATH_MODULES
    for SBTR in $SUBMODULE_TERRAFORM
    do
      get_time
      print_line2
      echo "$COL_GREEN[ $DATE ]       git@github.com:terraform-aws-modules/$SBTR.git \n"
      print_line2
      git submodule add git@github.com:terraform-aws-modules/$SBTR.git
      echo ""
    done
}

main() {
  header
  submodule_terrafom
  footer
}

### START HERE ###
main $@
